src/python/gudhi/sklearn/rips_persistence.py

```diff
@@ -0,0 +1,148 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Vincent Rouvreau
+#
+# Copyright (C) 2022 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from .. import RipsComplex
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import numpy as np
+# joblib is required by scikit-learn
+from joblib import Parallel, delayed
+
+# Mermaid sequence diagram - https://mermaid-js.github.io/mermaid-live-editor/
+# sequenceDiagram
+#     USER->>RipsPersistence: fit_transform(X)
+#     Note right of RipsPersistence: homology_dimensions=[i,j]
+#     RipsPersistence->>thread1: _tranform(X[0])
+#     RipsPersistence->>thread2: _tranform(X[1])
+#     Note right of RipsPersistence: ...
+#     thread1->>RipsPersistence: [array( Hi(X[0]) ), array( Hj(X[0]) )]
+#     thread2->>RipsPersistence: [array( Hi(X[1]) ), array( Hj(X[1]) )]
+#     Note right of RipsPersistence: ...
+#     RipsPersistence->>USER: [[array( Hi(X[0]) ), array( Hj(X[0]) )],<br/> [array( Hi(X[1]) ), array( Hj(X[1]) )],<br/> ...]
+
+
+class RipsPersistence(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing the persistence diagrams from a Vietoris-Rips complex.
+    """
+
+    def __init__(
+        self,
+        homology_dimensions,
+        max_edge_length=float('inf'),
+        input_type='point cloud',
+        nb_collapse=-1,
+        homology_coeff_field=11,
+        min_persistence=0.0,
+        expand_extra_dimension = True,
+        n_jobs=None,
+    ):
+        """
+        Constructor for the RipsPersistence class.
+
+        Parameters:
+            homology_dimensions (int or list of int): The returned persistence diagrams dimension(s).
+                Short circuit the use of :class:`~gudhi.representations.preprocessing.DimensionSelector` when only one
+                dimension matters (in other words, when `homology_dimensions` is an int).
+            max_edge_length (float): Rips value. Default is +Inf.
+            input_type (str): Can be 'point cloud' when inputs are point clouds, or 'lower distance matrix', when
```

Comment:
I think my main remaining hesitation is how much we want to split things into many different classes vs cram everything into a single class with a myriad of options. Trying to list some variants that may appear in the future
1. input is points (with an implicit or explicit distance function) / input is a graph
2. for graph input, it can be a dense or sparse matrix
2. for graph input, allow or not weights for the vertices (while remaining a flag filtration)
3. if input is points: compute the full distance matrix or use a kd-tree to find neighbors and get a sparse distance matrix (when a non-trivial threshold is specified)
4. sparse Rips (not super useful these days, while approximate, it tends to be worse than the exact edge collapse)
5. weighted Rips as in DTM filtrations: different enough to deserve its own class(es)